addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportDialog.java

```diff
@@ -281,6 +281,7 @@ public void changedUpdate(DocumentEvent e) {
     private JScrollPane getSectionsScrollPane() {
         if (sectionsPane == null) {
             sectionsPane = new JScrollPane();
+            sectionsPane.setMinimumSize((DisplayUtils.getScaledDimension(400, 300)));
```

Comment:
This is a workaround.

Reply:
See `validate()` and related methods.

Reply:
Sorry I'm still learning about Swing , and I'm not sure why setting the minimum size for a JPanel would be considered a workaround. Could you explain it to me, and maybe mention some other alternatives?

Reply:
Try resize the dialogue and see how it (mis)behaves, that's consequence of hardcoded sizes. We should not need to hardcode sizes, the panels have layouts so they are resized to fit the components as needed.

The fix is to call the `validate()` method after setting the new panel to the scroll pane, which allows the UI to layout the components again to fit the new components set.

Reply:
Cool now I learned something too. (Just gotta hope I remember it 😉)

Reply:
In this case a `repaint()` might be enough, since the `setViewportView(…)` already validates.

Reply:
Thanks for the information @thc202 . so i think calling sectionsPane.validate() will be enough, as it will validate the container and its children recursively.

Reply:
The problem is not with `sectionsPane` but the tab panel, try call `repaint()` after `getSectionsScrollPane().setViewportView(sectionPanel);`.

Reply:
ok repaint() is working fine. 
i have already google it, but still can you please explain me the difference between revalidate() and repaint() .

Reply:
Did you check the JavaDoc?